src/main/scala/no/ndla/searchapi/service/search/SearchConverterService.scala

```diff
@@ -880,58 +830,48 @@ trait SearchConverterService {
-                                                 resourceTypesWithParents,
-                                                 bundle))
-                } else {
-                  contextFilters.map(
-                    cf =>
-                      getSearchableTaxonomyContext(resource.id,
-                                                   pathIds,
-                                                   subject,
-                                                   relevanceId,
-                                                   relevance,
-                                                   contextType,
-                                                   List(cf),
-                                                   resourceTypesWithParents,
-                                                   bundle))
-                }
+          val contexts = topicsConnections.map({ tc =>
+            val relevanceId = tc.relevanceId.getOrElse("urn:relevance:core")
+            val relevanceName = bundle.relevances
+              .find(r => r.id == relevanceId)
+              .map(_.name)
+              .getOrElse("")
+            val relevance = SearchableLanguageValues(Seq(LanguageValue(DefaultLanguage, relevanceName)))
+
+            val topic = bundle.topics.find(topic => tc.topicid == topic.id)
+            topic
+              .map({
+                t =>
+                  getParentTopicsAndPaths(t, bundle, List(t.id)).flatMap({
+                    case (topic, topicPath) =>
+                      // Subjects needed to check visibility
+                      val subjectConnections = bundle.subjectTopicConnections.filter(_.topicid == topic.id)
+                      val subjects =
+                        bundle.subjects.filter(subject => subjectConnections.map(_.subjectid).contains(subject.id))
+
+                      val visibleSubjects = if (filterVisibles) {
+                        subjects.filter(_.metadata.forall(_.visible))
+                      } else {
+                        subjects
+                      }
+
+                      visibleSubjects.flatMap(subject => {
+                        val pathIds = (resource.id +: topicPath :+ subject.id).reverse
+                        List(
+                          getSearchableTaxonomyContext(resource.id,
+                                                       pathIds,
+                                                       subject,
+                                                       relevanceId,
+                                                       relevance,
+                                                       contextType,
+                                                       List.empty,
+                                                       resourceTypesWithParents,
+                                                       bundle))
+
+                      })
+                  })
               })
+              .getOrElse(List.empty)
```

Comment:
@jnatten Kunne denne vore skreve bedre? Kom gjerne med forslag.

Reply:
Hmm, jeg ser egentlig ikke noe problem med denne jeg. Gir det ikke mening at listen med contexter for `Topic`'en er en tom liste dersom vi ikke finner `Topic`'en?

Reply:
Eg berre tenkte det ikkje såg så elegant ut. Men det fungerer jo så kan la det være slk.